pkg/resolver/config.go

```diff
@@ -14,6 +30,134 @@ type Config struct {
 	Rewrites []RewriteConfig `yaml:"rewrite,omitempty" toml:",omitempty" json:",omitempty"`
 }
 
+// New creates a new [Resolver]
+func (rc Config) New(next resolver.Exchanger, log slog.Logger) (*Resolver, error) {
+	var err error
+
+	r := &Resolver{
+		debug:    make(map[string]slog.LogLevel),
+		log:      log,
+		name:     rc.Name,
+		suffixes: rc.Suffixes,
+
+		Next:      next,
+		Exchanger: resolver.ExchangerFunc(horizon.ForbiddenExchange),
+	}
+
+	switch {
+	case rc.Iterative:
+		err = rc.setupIterative(r, log)
+	case len(rc.Servers) > 0:
+		err = rc.setupForwarder(r, log)
+	default:
+		err = rc.setupChained(r, log)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+func (rc Config) setupIterative(r *Resolver, log slog.Logger) error {
+	var e resolver.Exchanger
+
+	switch {
+	case len(rc.Servers) > 0:
+		return &Error{
+			Resolver: rc.Name,
+			Reason:   "iterative resolver with specific servers not yet supported.",
+		}
+	case rc.Recursive:
+		return &Error{
+			Resolver: rc.Name,
+			Reason:   "iterative resolvers can't be recursive.",
+		}
+	}
```

Comment:
this could be a simple `if`:
```go
if len(rc.Servers)>0 || rc.Recursive {
    return &Error{
        Resolver: rc.Name,
        Reason: "iterative resolver does not support specifying servers or being recursive",
    }
}